docs/arcadia-users-group/20220919-aws-s3-cli/lesson.md

```diff
@@ -0,0 +1,301 @@
+* `<SOURCE_PATH>` and `<TARGET_PATH>` could be a path to a local file/directory, or an S3 file/directory path. For local files, the paths can be absolute or relative.
+
+
+As an example, something like `aws s3 cp lesson.md s3://aug-workshop-demo/lesson.md` would copy the local `lesson.md` file from my computer to the S3 bucket `aug-workshop-demo` and create a file called `lesson.md`.
+
+## S3 commands we'll work with
+
+For all the following commands, we'll be working with an S3 bucket called `aug-workshop-demo` hosted through Arcadia's AWS account. For the workshop, we'll focus on a small but useful subset of commands, but you can find the full reference documentation [here](https://docs.aws.amazon.com/cli/latest/reference/s3/).
+
+### ls
+
+In the [first shell workshop](../20220906-intro-to-shell1/lesson.md), we learned that `ls` lists contents of a directory (**l**i**s**t). It serves the same function for the S3 CLI.
+
+Let's start with listing all the Arcadia Science S3 buckets:
+
+```{bash}
+aws s3 ls
+```
+
+Now let's explore what one of these buckets looks like, starting with the `aug-workshop-demo` bucket.
+
+```{bash}
+aws s3 ls s3://aug-workshop-demo
+```
+
+If everything went correctly, you should see the `demo` folder under the `aug-workshop-demo` bucket.
+
+### cp
+
+In the [second shell workshop](../20220912-intro-to-shell2/lesson.md), we learned that `cp` copies a file or directory. The S3 command also serves a similar purpose. Let's try it out by trying to copy the file we created earlied `copy.txt` to the S3 bucket under a folder with your initials. For me, this would be in `s3://aug-workshop-demo/fmc/`.
+
+```{bash}
+aws s3 cp copy.txt s3://aug-workshop-demo/fmc/copy.txt
+```
+
+If you ignore the last part of the S3 path, it'll create a file with the same name:
+
+```{bash}
+aws s3 cp copy.txt s3://aug-workshop-demo/fmc/
+```
+
+Alternatively, you can give a new name to the file on S3:
+
+```{bash}
+aws s3 cp copy.txt s3://aug-workshop-demo/fmc/new-copy.txt
+```
+
+Let's see the changes:
+```{bash}
+ls
+```
+The file is still in our directory, it hasn't moved.
+
+```{bash}
+aws s3 ls s3://aug-workshop-demo
+aws s3 ls s3://aug-workshop-demo/fmc/
+```
+
+Now let's try copying the `sync` directory by:
+```{bash}
+aws s3 cp sync/ s3://aug-workshop-demo/fmc/
+```
+
+That failed! This is because the `cp` command directly works with individual files. For copying folders, we need to use the `--recursive` flag. So, let's try this:
+
+```{bash}
+aws s3 cp --recursive sync/ s3://aug-workshop-demo/fmc/
+aws s3 ls s3://aug-workshop-demo/fmc/
+```
+
+### mv
+
+In the [second shell workshop](../20220912-intro-to-shell2/lesson.md), we learned that `mv` moves or renames a file or directory. The S3 command also serves a similar purpose. Let's start with a moving example by trying to move the `move.txt` file we created earlier to the S3 bucket under a folder with your name. For me, this would be in `s3://aug-workshop-demo/mert/`.
```

Comment:
How come a folder with your name now instead of initials?

Reply:
oh i see, because you're going to move the file from your name dir to your initial dir later. Got it!